### PR TITLE
Feature/image quality

### LINKED ISSUE #52: Grabber image quality configuration
Might still need to determine the best way to represent this as a setting... This should be very simple for the end user to figure out, and not something they need to guess at.

A good way may be to simply have the user note how many LEDs wide & tall their setup is. Seems like that should give us a good bitrate range to aim for when scaling the captured image from the device running the grabber.

### PR DESCRIPTION
**Pull request by abrenoch**
Adds the ability to configure the quality of the grabbed image.

Decided to approach this using the LED count of the device hyperion is driving - those values are used to determine the optimal quality for the image. Since there are only specific values we can scale the screen by (while maintaining correct aspect ratio), this seems like the most user friendly way to pick the optimal setting.

There is a small change to the notification buttons in `HyperionScreenService` which I'm not sure how got mixed into this commit, but that is just adding the "EXIT" text to the strings xml & removing the extra button it is initialized with but never shows for some reason. So nothing that should impact the functionality of the app.

Closes #52!

### CODE REVIEW

File: common/src/main/java/com/abrenoch/hyperiongrabber/common/util/HyperionGrabberOptions.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        FRAME_RATE = frameRate;
+
+        if (DEBUG) {
+            Log.d(TAG, "Horizontal LED Count: " + String.valueOf(horizontalLED));
+            Log.d(TAG, "Vertical LED Count: " + String.valueOf(verticalLED));
+            Log.d(TAG, "Minimum Image Packet: " + String.valueOf(MINIMUM_IMAGE_PACKET_SIZE));
+        }
+    }
+
+    // just returns the frame rate
```
Comment (ninovanhooff) [inline common/src/main/java/com/abrenoch/hyperiongrabber/common/util/HyperionGrabberOptions.java]: Love the elaborate comments in this file, but when a comment starts with "just", you are probably overdoing it :-)

File: common/src/main/java/com/abrenoch/hyperiongrabber/common/util/HyperionGrabberOptions.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    public int getFrameRate() { return FRAME_RATE; }
+
+    // returns the divisor best suited to be used to meet the minimum image packet size
+    //      Since we only want to scale using whole numbers, we need to find what common divisors
+    //      are available for the given width & height. We will check those divisors to find the smallest
+    //      number (that we can divide our screen dimensions by) that would meet the minimum image
+    //      packet size required to match the count of the LEDs on the destination hyperion server.
+    public int findDivisor(int width, int height) {
```
Comment (ninovanhooff) [inline common/src/main/java/com/abrenoch/hyperiongrabber/common/util/HyperionGrabberOptions.java]: When commenting on fields and functions / methods. Use documentation comments
https://www.tutorialspoint.com/java/java_documentation.htm

File: common/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -33,6 +36,16 @@
...
 
+    <string name="pref_key_y_led">pref_key_y_led</string>
+    <string name="pref_title_y_led">Vertical LED Count</string>
+    <string name="pref_summary_y_led">The vertical count of LEDs on your hardware</string>
+    <integer name="pref_default_y_led">20</integer>
+
+    <string name="pref_key_x_led">pref_key_x_led</string>
+    <string name="pref_title_x_led">Horizontal LED Count</string>
+    <string name="pref_summary_x_led">The horizontal count of LEDs on your hardware</string>
+    <integer name="pref_default_x_led">36</integer>
```
Comment (ninovanhooff) [inline common/src/main/res/values/strings.xml]: I feel this default is too specific. Not exactly sure where the 60x60 came from, but if we have no clue about the users's situation, let's guess on the high side and go with that. Or go with 60x34, which is 60 wide in a 16:9 aspect

File: common/src/main/java/com/abrenoch/hyperiongrabber/common/util/HyperionGrabberOptions.java (hunk 2 above)
Comment (abrenoch) [reply common/src/main/java/com/abrenoch/hyperiongrabber/common/util/HyperionGrabberOptions.java]: The worst part is I know this and use that constantly in my day job... I'm not sure why I made those like that 🤡

File: common/src/main/res/values/strings.xml (hunk 3 above)
Comment (abrenoch) [reply common/src/main/res/values/strings.xml]: Sure we can go with whatever sounds the most reasonable. I just used the count from my setup. 60x34 sounds perfectly fine to me

File: common/src/main/java/com/abrenoch/hyperiongrabber/common/util/HyperionGrabberOptions.java (hunk 1 above)
Comment (abrenoch) [reply common/src/main/java/com/abrenoch/hyperiongrabber/common/util/HyperionGrabberOptions.java]: That one did seem out of place :).

I'll get these comment blocks updated!

**Comment by Paulchen-Panther** [comment]
I hope this fixes the [OOM issue](https://github.com/abrenoch/hyperion-android-grabber/issues/68)

**Comment by abrenoch** [comment]
I was still running into the OOMs with @Paulchen-Panther 's fix (despite my early enthusiasm hah). This fix sets a maximum response size and sets it to 1KB. The typical response size is 4 bytes, so if it is anything larger than that something fishy is up, so a 1KB maximum should be acceptable.

This seems to make things a lot more stable when performing the 'hammer the toggle button as fast as I can' test 😁

**Comment by Paulchen-Panther** [comment]
I used the variable usedSocket to counteract the socket closed error.

**Comment by abrenoch** [comment]
Unfortunately I'm still seeing the `E/HyperionScreenService: Socket closed` error happening with those changes, so I'm not sure it is having the intended effect. I honestly don't think it is a big deal anyway, since that error appears to be handled gracefully within `HyperionScreenService` and was unrelated to the OOM error (seemingly anyways).

Likewise with the image buffer error I mentioned earlier, I don't really think it is going to cause any problems since the error is being handled. The only concern I had there was it potentially not closing out an image from the buffer, but those errors appear to only be presented once the image is already closed (which is probably more of a synchronization issue).

I think if this PR looks alright as-is then we should merge and address any more fixes in separate PRs... I just don't want the scope of this one to start creeping if these issues existed already anyway!

**Comment by abrenoch** [comment]
The OOM issue seems to have been resolved by using a `BufferedByteArray` when reading the socket response

**Pull request merged**